client/server/daemon.go

```diff
@@ -0,0 +1,78 @@
+package server
+
+import (
+	"context"
+	"fmt"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/wiretrustee/wiretrustee/client/internal"
+	"github.com/wiretrustee/wiretrustee/client/proto"
+)
+
+// Server for service control.
+type Server struct {
+	managementURL string
+	configPath    string
+	stopCh        chan int
+	cleanupCh     chan<- struct{}
+
+	config *internal.Config
+	proto.UnimplementedDaemonServiceServer
+}
+
+// New server instance constructor.
+func New(managementURL, configPath string, stopCh chan int, cleanupCh chan<- struct{}) *Server {
+	return &Server{
+		managementURL: managementURL,
+		configPath:    configPath,
+		stopCh:        stopCh,
+		cleanupCh:     cleanupCh,
+	}
+}
+
+// Login uses setup key to prepare configuration for the daemon.
+func (s *Server) Login(ctx context.Context, msg *proto.LoginRequest) (*proto.LoginResponse, error) {
+	config, err := internal.GetConfig(s.managementURL, s.configPath, msg.PresharedKey)
+	if err != nil {
+		return nil, err
+	}
+	s.config = config
+
+	// login operation uses backoff scheme to connect to management API
+	// we don't wait for result and return response immediately.
+	go func() {
+		if err := internal.Login(s.config, msg.SetupKey); err != nil {
+			log.Errorf("failed login: %v", err)
+		}
+	}()
+
+	return &proto.LoginResponse{}, nil
+}
+
+// Up starts engine work in the daemon.
+func (s *Server) Up(ctx context.Context, msg *proto.UpRequest) (*proto.UpResponse, error) {
+	if s.config == nil {
+		return nil, fmt.Errorf("config is not defined, please call login command first")
+	}
+
+	// connect operation uses backoff scheme to grab configuration from management API
+	// we don't wait for result and return response immediately.
+	go func() {
```

Comment:
There will be a problem when we trigger UP command twice with a wrong key. 

I think that users will run `wiretrustee up` just by mistake without a setup key.
The daemon will enter into a backoff loop because management will return error (not registered)
Down won't help because the stop channel won't be read at this point. 
Consequent runs of `wiretrustee up` will cause a second loop of RunClient

Reply:
Similarly, 
When UP command was called properly (with a setup key), consequent UP will trigger a second RunClient() causing errors.

**There must be some state check if service already runs or not.**

To reproduce:
1. clean run (delete existing config)
2. start wiretrustee service
3. run `wiretrustee up --setup-key <KEY>`
4. check logs:
```
time="2022-02-20T18:42:27+01:00" level=info msg="generating new config /etc/wiretrustee/config.json"
time="2022-02-20T18:42:27+01:00" level=error msg="peer registration required. Please run wiretrustee login command first"
time="2022-02-20T18:42:27+01:00" level=warning msg="rpc error: code = PermissionDenied desc = provided peer with the key wgPubKey ekQAxUev21pszxop/vP+KT1r0324N0Hkqv4SWeFkHVQ= is not registered"
time="2022-02-20T18:42:27+01:00" level=info msg="peer has been successfully registered on Management Service"
time="2022-02-20T18:42:29+01:00" level=info msg="connected to the Signal Service stream"
time="2022-02-20T18:42:29+01:00" level=info msg="Wiretrustee engine started, my IP is: 100.64.0.6/24"
time="2022-02-20T18:42:29+01:00" level=info msg="connected to the Management Service stream"
time="2022-02-20T18:42:30+01:00" level=info msg="connected to peer qDZ5IZeRjEfKHS+H6LvPJy6iB6IOfKnjFurwFovAMSw= [laddr <-> raddr] [89.12.99.214:50702 <-> 34.159.113.137:33605]"
time="2022-02-20T18:42:31+01:00" level=info msg="connected to peer m7bmlo4gN7GjmJMRPgi9CllB7v2Ua8uQ+TjnWE7b61w= [laddr <-> raddr] [172.17.0.1:40319 <-> 34.141.3.134:60077]"
```
5.  run `wiretrustee up --setup-key <KEY>`
6.  errors will occur, check logs: 
```
time="2022-02-20T18:42:34+01:00" level=info msg="peer has successfully logged-in to Management Service"
time="2022-02-20T18:42:34+01:00" level=info msg="connected to the Signal Service stream"
time="2022-02-20T18:42:34+01:00" level=info msg="Wiretrustee engine started, my IP is: 100.64.0.6/24"
time="2022-02-20T18:42:34+01:00" level=info msg="connected to the Management Service stream"
time="2022-02-20T18:42:34+01:00" level=error msg="Management stream has been closed by server: EOF"
time="2022-02-20T18:42:36+01:00" level=info msg="connected to the Management Service stream"
time="2022-02-20T18:42:36+01:00" level=error msg="Management stream has been closed by server: EOF"
time="2022-02-20T18:42:36+01:00" level=info msg="connected to the Management Service stream"
time="2022-02-20T18:42:37+01:00" level=error msg="Management stream has been closed by server: EOF"
time="2022-02-20T18:42:38+01:00" level=info msg="connected to the Management Service stream"
time="2022-02-20T18:42:38+01:00" level=error msg="Management stream has been closed by server: EOF"

```